### PR TITLE
6914 Ext4 - Add implicit sparse blocks to end of file

### PR DESCRIPTION
**Pull request by APriestman**

### CODE REVIEW

**Comment by esaunders** [comment]
Looks like Codacy is pointing out that while there is a NULL check on fs_attr on line 1598, there is an attempt to de-reference it on line 1596.

**Comment by APriestman** [comment]
Weird - I thought I went in and fixed that. Maybe I never pushed it up. I'll change it.

File: tsk/fs/ext2fs.c [hunk 1]
```diff
@@ -1585,6 +1585,49 @@ ext2fs_extent_tree_index_count(TSK_FS_INFO * fs_info,
     return count;
 }
 
+/** \internal
+* If the file length is longer than what is in the attr runs, add a sparse
+* data run to cover the rest of the file.
+*
+* @return 0 if successful or -1 on error.
```
Comment (esaunders) [inline tsk/fs/ext2fs.c:1592]: Looks like only 0 or 1 get returned.

File: tsk/fs/ext2fs.c [hunk 2]
```diff
@@ -1585,6 +1585,49 @@ ext2fs_extent_tree_index_count(TSK_FS_INFO * fs_info,
...
+            return 1;
+        }
+        data_run->offset = end_of_runs;
+        data_run->addr = 0;
+        data_run->len = total_file_blocks - end_of_runs;
+        data_run->flags = TSK_FS_ATTR_RUN_FLAG_SPARSE;
+
+        // Save the run.
+        if (tsk_fs_attr_add_run(fs_info, fs_attr, data_run)) {
+            return 1;
```
Comment (esaunders) [inline tsk/fs/ext2fs.c:1626]: Do you need to free the attr run here?

**Review by esaunders — APPROVED** [review]

**Pull request merged**